### ISSUE TITLE
mypaint has file conflict with libmypaint

### ISSUE THREAD

**Issue by jbicha** [mypaint/mypaint#918]
#### Description of the problem

Hi, I recently packaged libmypaint 1.3.0 for Debian and Ubuntu as a required dependency for gimp 2.10 RC1.

I just noticed that there is a file conflict. Both libmypaint and mypaint ship the translation files named `libmypaint.mo`

What do you recommend we do here?

1. Is it practical to package a newer version of mypaint that uses the shared libmypaint?

2. Should I disable translations in libmypaint for now? Are any of those translatable strings exposed in gimp's user interface?

#### Basic system details

MyPaint version: 1.2.0
Operating system: Ubuntu 18.04

**Comment by achadwick** [comment]
1. Yes, eminently practical for Debian testing/unstable! I'd love to see this in the OS I use, but I haven't had the time to do it myself. We have an example here: <https://github.com/mypaint/mypaint.deb> (maintained as a separate repository to avoid breaking the Debian Python packaging team's workflow; there's one for libmypaint too). Would you like access?

2. Please don't disable translations in libmypaint. They will be exposed in MyPaint's user interface (and possibly GIMP's).

**Comment by jbicha** [comment]
@achadwick Are you recommending that all the Linux distros switch to a git snapshot of mypaint then?

Do you have any schedule for when the next major mypaint release will be?

Are there any major issues I should know about in git master?

Ubuntu 18.04 LTS will release in 3 weeks…

**Comment by achadwick** [comment]
May be worth calling it soon then, given that timescale.

There have been a few Wayland issues. I'm hoping they're fixed now, though.

**Comment by darix** [comment]
openSUSE would appreciate a release as well. (for the same reasons)

**Comment by badshah400** [comment]
We have run into a similar issue for openSUSE too. It has become necessary to have libmypaint-devel (by which I mean the header files and so on needed for coding against libmypaint) for gimp 2.10, while mypaint-devel 1.1.x also provides the same files, leading to file conflicts between the two packages. Fwiw, on openSUSE we don't package mypaint 1.2.x because of dependency issues (we miss pygtkcompat), and would rather move on to mypaint 1.3.0 directly.

It would be helpful for us downstream packagers to know the current state of git master. Would it be ok to package a snapshot of git master and replace existing installs of mypaint on users' computers with that? Will there be a 1.3.0-beta tagged release at some point, or perhaps 1.3.0 release itself is imminent? Thanks for any clarifications.

**Comment by darkshram** [comment]
I have already done gimp 2.10.0 packages for my Linux distro and they will be released this weekend. Would appreciate to know the status for git master too. We ship both gimp and mypaint (my son uses mypaint and really loves it), and we are considering to **temporarily** obsolete mypaint 1.2.x with libmypaint 1.3.0 —for a clean update— considering the relevance of gimp 2.10.0. For the time being we will be suggesting our users to install mypaint using flatpak until we can ship mypaint 1.3.x or until I figure out how to repackage mypaint to avoid conflicts.

**Comment by nyov** [comment]
@jbicha, would you be able to follow up on https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=894757 please? Is there a workaround or not, since upstream seems to be asleep on the matter? That'd be awesome, thanks!

**Comment by darix** [comment]
any progress?

**Comment by nyov** [comment]
For debian yes, in as far as Gimp 2.10 is now in testing (using libmypaint), and there is simply a conflict with mypaint which doesn't allow the installation of mypaint in parallel with gimp/libmypaint.

Considering the user-base of Gimp in debian is about an order of magnitude higher than mypaint (~11000 vs ~1200), that seems like a decent solution until mypaint fixes itself.

**Comment by mariodebian** [comment]
Hi.

I made a patch for mypaint/libmypaint to rename locale domain and install with new Gimp 2.10


[052-rename-locale-mypaint12.patch.txt](https://github.com/mypaint/mypaint/files/2522100/052-rename-locale-mypaint12.patch.txt)

**Comment by QuLogic** [comment]
Fedora 29 is also running into this conflict. It also breaks updates from 28 to 29 if both are installed before hand (which can be worked around, but is annoying.)

**Comment by BrawlPups** [comment]
It is now December 2018 and the lack  of a solution still exists.

**Comment by briend** [comment]
New release should be soon!

**Comment by zuzzurro** [comment]
Same here. Fedora 29 and setup broken. Is the new release available somewhere as a beta package?

**Comment by darix** [comment]
is there anything we can do to help with the issue?

**Comment by briend** [comment]
@darix yes, you could test out the 2.0 alpha versions and help me identify any other glaring bugs that I've introduced! :)

**Comment by LaurianoElmiroDuarte** [comment]
Any solution ? 
i'm having some problem.
my system is fedora 29.

**Comment by aoloe** [comment]
i have seen several project die, with the latest release being an alpha candidate...

please, consider "rushing out" a 2.0, even if it has bugs you don't like, and avoid mypaint being kicked from all linux distribution because of the gimp incompatibility.

**Comment by bergamote** [comment]
So just checking, are we still in some kind of catch 22 situation here?

Mypaint is by far my favourite program to draw with, but it becomes near useless if you can't use it in conjunction with gimp, and on Ubuntu 19 they still can't be installed at the same time. 
Then again, MX Linux managed to work around the dependency clash so, is there any way of doing the same for other distros?

**Comment by aoloe** [comment]
@bergamote more and more grateful for probono's work on appimage: you can downlowad the appimage, make it executable, and put it in `/usr/local/bin/`...

it will use more resources than programs usually do on linux, but if your computer is somehow powerful enough, it's a good workaround.

this having been said, i'm very glad to see a new alpha release, but i fear that it still will take lot of time until a new stable release will be out...
... even if the 2.0 "milestone" only lists two tasks to be done until 2.0 can be released. both relating to the way the brushes are stored on disk.

i'd really like to see a 2.0 release soon or at least 2.0alpha being packaged for debian / ubuntu as the new mypaint (this is what will probably happen to scribus if 1.6 is not released before the end of the year...)

**Comment by eugeneloza** [comment]
@aoloe thanks a lot for the hint! I've never used those before, but now finally MyPaint is again working for me! :)

For those like me who don't know what an AppImage is - https://appimage.org/ tl;dr - it's an executable with all dependencies included.

For those like me who don't know were to get MyPaint AppImage - https://github.com/mypaint/mypaint/releases and look in `assets`

Thanks again!

**Comment by jplloyd** [comment]
@aoloe 
Basically the situation is this: since the handling/rendering of image data has changed between 1.2 and 2.0 we need to think about (and implement) a solution to backwards compatibility. There are a couple more things I would like to get in before the beta phase, but that one is the most pressing.

If all MyPaint users are fine with all of their old .ora files not looking the same in 2.0 as they did in 1.2, we could release tomorrow (with incomplete translations and missing documentation), but I doubt that's the case. The rough plan is to implement a legacy mode that supports the 1.2 workflow and rendering, and defaulting to that mode when opening files which cannot be recognized as 2.0 files.
It will also be possible to turn off legacy mode manually for old files (which is necessary for the new pigment blending to work properly), and when these files are subsequently saved, store information about which mode to use the next time it is opened.

This is not a huge amount of work, and a lot of it is already done, but it will take a little while to get it fully ready. My _hope_ is that we can go into beta in early December and aim for a release in January 2020.

@eugeneloza 

About a month ago I set up automated releases of mypaint appimages, if you want to check out the bleeding edge between releases: https://github.com/mypaint/mypaint-appimage/releases/tag/continuous

This is just for future reference. There have been no user-visible changes since the last alpha.

**Comment by bergamote** [comment]
Thanks all!! I'm mypainting again!

**Comment by aoloe** [comment]
@jplloyd good luck for the upcoming release!

**Comment by jplloyd** [comment]
MyPaint 2.0 is now released: https://github.com/mypaint/mypaint/releases/tag/v2.0.0
Hopefully without too many issues (definitely no libmypaint conflicts at least).
